### DISCUSSION TITLE
Global keyboard hook in OBS plugin problem

### DISCUSSION THREAD

**Discussion by SetoKaiba** [obsproject/obs-studio#8748] [Development Questions]
![image](https://user-images.githubusercontent.com/159540/232177467-11fea569-3db8-4d22-b1d4-df416ff8f1bf.png)
Hi, I'm developing a live metaverse. I want to hook the global keyboard.

But the keyboard hook is working only with OBS focused.
The mouse hook is working with OBS in the background.

![image](https://user-images.githubusercontent.com/159540/232177552-56500c48-4eb1-43e4-9534-efe8757f6f15.png)
I want to implement such feature like BongoCat which show the broadcaster's keyboard and mouse state.
But I want to show it in the 3d space. So I can't just broadcast it as window capture.
Instead I have to capture the keyboard mouse state and add it to the stream to sync with the video and audio.
I have to modify the obs plugin to support it.

![image](https://user-images.githubusercontent.com/159540/232177629-c8a9e0e6-0b03-464a-9790-7b2e82b560ba.png)
Keyboard state is zero if OBS is in the background.
Why is global keyboard hook is not working?

**Comment by Fenrirthviti** [comment]
This appears to be a website, and not an OBS plugin like Bongocat is. I'm confused what this has to do with OBS.

**Reply by SetoKaiba** [reply]
It's my website. I use WebRTC whip PR to send the keyboard and mouse state via RTP. The problem is I can't global hook keyboard with OBS plugin. But the mouse can be hooked globally. The keyboard hook is working while OBS in the frontend.

**Reply by SetoKaiba** [reply]
https://github.com/obsproject/obs-studio/pull/7926

**Reply by Fenrirthviti** [reply]
That PR has not been merged, and is not supported at this time. In addition, I don't believe that kind of method being used to send keystrokes is something we plan to support. I would recommend trying the #plugins-and-tools channel on our Discord if someone has any experience in this area, but this isn't really related to anything that we currently support.

**Reply by SetoKaiba** [reply]
I know that it's not related to obs existing feature. I'm just ask for help that why global hook is only working when it's in frontend. It's maybe related to the plugin design.
Previously I considered use a standalone program to send the keystroke. But the problem is that I can't sync the input with the video. So I have to do it in OBS. I'm planning to use this to implement some kind of esports broadcast in my live metaverse. I'm try to show the virtual 3d player's activity base on the keystroke.

**Reply by SetoKaiba** [reply]
@Fenrirthviti I saw that you close this discussion. Can this be an open discussion? Maybe I can get help from others.
